src/common/procedure/src/local/runner.rs

```diff
@@ -288,26 +301,48 @@ impl Runner {
                             return;
                         }
 
+                        // Cleans poisons before persist.
+                        if status.need_clean_poisons() {
+                            if let Err(e) = self.clean_poisons().await {
+                                error!(e; "Failed to clean poison for procedure: {}", self.meta.id);
+                                // TODO(weny): Finds a better way to handle the retry.
+                                return;
```

Comment:
Is it possible to set the state to retrying?

Reply:
resolved in [3331fce](https://github.com/GreptimeTeam/greptimedb/pull/5822/commits/3331fce63408d6f5967551f116194a93890b094c)